src/map/Map.tsx

```diff
@@ -1,65 +1,130 @@
+import ReactMapGL, { MapEvent, Popup, WebMercatorViewport } from 'react-map-gl'
+import 'mapbox-gl/dist/mapbox-gl.css'
 import { Coordinate, QueryPoint } from '@/stores/QueryStore'
-import React, { useEffect, useRef, useState } from 'react'
-import styles from '@/map/Map.module.css'
-import Mapbox, { ViewPort } from '@/map/Mapbox'
+import React, { useState } from 'react'
 import Dispatcher from '@/stores/Dispatcher'
-import { MapIsLoaded } from '@/actions/Actions'
-import { Bbox, Path } from '@/api/graphhopper'
-import { StyleOption } from '@/stores/MapOptionsStore'
-import { PathDetailsPoint } from '@/stores/PathDetailsStore'
+import { MapIsLoaded, SetViewport } from '@/actions/Actions'
+import { RasterStyle, StyleOption, VectorStyle } from '@/stores/MapOptionsStore'
+import { ViewportStoreState } from '@/stores/ViewportStore'
+import { PopupComponent } from '@/map/Popup'
+import { MapLayer } from '@/layers/MapLayer'
 
 type MapProps = {
-    selectedPath: Path
-    paths: Path[]
+    viewport: ViewportStoreState
     queryPoints: QueryPoint[]
-    bbox: Bbox
     mapStyle: StyleOption
-    pathDetailPoint: PathDetailsPoint | null
-    highlightedPathDetailSegments: Coordinate[][]
+    mapLayers: MapLayer[]
 }
 
-export default function ({
-    selectedPath,
-    paths,
-    queryPoints,
-    bbox,
-    mapStyle,
-    pathDetailPoint,
-    highlightedPathDetailSegments,
-}: MapProps) {
-    const mapContainerRef: React.RefObject<HTMLDivElement> = useRef(null)
-    const [map, setMap] = useState<Mapbox | null>(null)
-    const prevViewPort = useRef<ViewPort | null>(null)
-    useEffect(() => {
-        prevViewPort.current = null
-    }, [bbox])
+export default function ({ viewport, mapStyle, queryPoints, mapLayers }: MapProps) {
+    const [popupCoordinate, setPopupCoordinate] = useState<Coordinate | null>(null)
+    const longTouchHandler = new LongTouchHandler(e => setPopupCoordinate({ lng: e.lngLat[0], lat: e.lngLat[1] }))
```

Comment:
I guess this could be modeled with an effect now. I chose to model this as a class because this Handler lived outside of a react component. We can also leave it like this for now. 

I'm not sure though what happens if a touch down event happens and the component gets re-rendered, before the timout finishes. The current handler implementation doesn't have a cancel function

Reply:
Maybe we can continue here in #111.

Reply:
Ah, yes. I didn't notice this issue yet